code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingFullscreenMessageDelegate.java

```diff
@@ -89,63 +89,70 @@ public boolean overrideUrlLoad(final FullscreenMessage fullscreenMessage, final
             return true;
         }
 
-        // check adbinapp scheme
+
         final String messageScheme = uri.getScheme();
 
+        // Quick bail out if scheme is not "adbinapp"
         if (messageScheme == null || !messageScheme.equals(MessagingConstants.QueryParameters.ADOBE_INAPP)) {
             Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "Invalid message scheme found in URI. (%s)", urlString);
             return false;
         }
 
-        // url decode the query parameters
-        final String queryParams;
-        try {
-            queryParams = URLDecoder.decode(uri.getQuery(), StandardCharsets.UTF_8.toString());
-        } catch (final UnsupportedEncodingException exception) {
-            Log.debug(MessagingConstants.LOG_TAG, SELF_TAG,  "UnsupportedEncodingException occurred when decoding query parameters %s.", uri.getQuery());
-            return false;
-        }
-
-        // Populate message data
-        final Map<String, String> messageData = extractQueryParameters(queryParams);
-
         final MessageSettings messageSettings = fullscreenMessage.getMessageSettings();
         final Message message = (Message) messageSettings.getParent();
 
-        if (!MapUtils.isNullOrEmpty(messageData)) {
-            // handle optional tracking
-            final String interaction = messageData.remove(MessagingConstants.QueryParameters.INTERACTION);
-            if (!StringUtils.isNullOrEmpty(interaction)) {
-                // ensure we have the MessagingExtension class available for tracking
-                final Object messagingExtension = message.getParent();
-                if (messagingExtension != null) {
-                    Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "Tracking message interaction (%s)", interaction);
-                    message.track(interaction, MessagingEdgeEventType.IN_APP_INTERACT);
-                }
+        // Handle query parameters
+        final String queryString = uri.getQuery();
+        if (!StringUtils.isNullOrEmpty(queryString)) {
```

Comment:
null check the decoded querystring before decoding

Reply:
uri.getQuery is returning url decoded query params. we can simplify this further by eliminating the url decoder usage.